lib/galaxy/jobs/__init__.py

```diff
@@ -1883,14 +1883,14 @@ def get_session_id(self):
     def get_env_setup_clause(self):
         if self.app.config.environment_setup_file is None:
             return ''
-        return '[ -f "{}" ] && . {}'.format(self.app.config.environment_setup_file, self.app.config.environment_setup_file)
+        return f'[ -f "{self.app.config.environment_setup_file}" ] && . {self.app.config.environment_setup_file}'
 
     def get_input_dataset_fnames(self, ds):
         filenames = []
         filenames = [ds.file_name]
         # we will need to stage in metadata file names also
         # TODO: would be better to only stage in metadata files that are actually needed (found in command line, referenced in config files, etc.)
-        for key, value in ds.metadata.items():
+        for value in ds.metadata.values():
```

Comment:
```suggestion
        for value in ds.metadata.values():
```
ds.metadata is  MetadataCollection instance and doesn't have a values() method. MetadataCollection overwrites __getattr__ to return None if the attribute is not found, so that explains the framework test failure. To be on the safe side I'd suggest
```suggestion
        for _, value in ds.metadata.items():
```
should there ever be a metadata element called values. Otherwise I guess we could add a values method ?

Reply:
Thanks for looking into that, I debugged it this morning and came to the same conclusions. Fixed in a rebase, I'm looking into implement a proper `Mapping`.